lib/v2/codeforces/contests.js

```diff
@@ -0,0 +1,37 @@
+const got = require('@/utils/got');
+const path = require('path');
+const { art } = require('@/utils/render');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+const contestAPI = 'https://codeforces.com/api/contest.list';
+
+module.exports = async (ctx) => {
+    const contestsData = await got.get(contestAPI).json();
+    const contests = contestsData.result;
+
+    const items = contests
+        .filter((contests) => contests.phase === 'BEFORE')
+        .map((contest) => {
+            const title = String(contest.name);
+            const description = art(path.join(__dirname, 'templates/contest.art'), {
+                title,
+                startTime: timezone(parseDate(contest.startTimeSeconds * 1000), +0),
+                durationTime: contest.durationSeconds / 60,
+                relativeTime: Math.floor(contest.relativeTimeSeconds / 60),
+                type: contest.type,
+            });
+
+            return {
+                title,
+                description,
+                link: 'https://codeforces.com/contests',
```

Comment:
Thank you very much for help me for this Route Request. one suggestion is that if the link for each item can navigate me to the direct contest page. That will be much helpful.

```suggestion
                link: 'https://codeforces.com/contests/' + contest.id,
```

Reply:
OK